w10h01/test/pgdp/trials/TrialOfTheDreamsTest.java

```diff
@@ -0,0 +1,89 @@
+package pgdp.trials;
+
+import java.util.function.Function;
+import java.security.SecureRandom;
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+import java.time.Duration;
+import static org.junit.jupiter.api.Assertions.assertTimeoutPreemptively;
+
+public class TrialOfTheDreamsTest {
+
+	// change for harder decoding
+	static final int keyLen = 16;
+	// change to test expected code duration in seconds (Artemis times out after 5s)
+	static final int expectedTime = 5;
+	// change to test a specific key with "test_not_random_LockPick"
+	static final byte[] expectedKey = new byte[] { 40, 41, 42, 43 };
+	// Use this seed in Random
+	static final int seed = 420;
+
+	@Test
+	public void testRandomLockPick() {
+		assertTimeoutPreemptively(Duration.ofSeconds(expectedTime), () -> {
+
+			ByteArrayGenerator lock = new ByteArrayGenerator(keyLen);
+			byte[] keyExpected = lock.generatedArray;
+			byte[] keyActual = TrialOfTheDreams.lockPick(lock);
+
+			assertArrayEquals(keyExpected, keyActual);
+			Thread.sleep(500);
+
+		});
+	}
+
+	@Test
+	public void test_not_random_LockPick() {
+		assertTimeoutPreemptively(Duration.ofSeconds(expectedTime), () -> {
+			Function<byte[], Boolean> lock = key -> Arrays.equals(key, expectedKey);
+			byte[] result = TrialOfTheDreams.lockPick(lock);
+			assertArrayEquals(expectedKey, result);
+
+			Thread.sleep(500);
```

Comment:
What is this for? #_2_

```suggestion
```

Reply:
To ensure system is no longer busy (better execution of next test)
You could theoratically remove it, I guess :p